samples/subsys/canbus/isotp/Kconfig

```diff
@@ -10,4 +10,19 @@ config SAMPLE_LOOPBACK_MODE
 	help
 	  Set the CAN controller to loopback mode. This allows testing without a second board.
 
+config SAMPLE_CAN_FD_MODE
+	bool "Use CAN-FD"
+	select CAN_FD_MODE
+	select ISOTP_USE_CAN_FD
+
+# A frame has 7 bytes payload, we are using a BS of 8 and need one char for the
+# string termination (7 * 8 + 1 = 57)
+config ISOTP_RX_BUF_SIZE
+	default 63 if SAMPLE_CAN_FD_MODE
```

Comment:
I think this can be simplified by using format precision to print the non-null terminated strings:

    printk("%.*s", buf->len, buf->data);

Although, I think some of the minimal string formatting libraries don't support precision.
`k_str_out` seems to be a good alternative:

    k_str_out(buf->data, buf->len);

Reply:
As far as I can see we don't even need it as the terminating null character is also sent over ISO-TP (using `sizeof` and not `strlen` in the send function).

Reply:
It's the receive side that is the issue (`rx_8_0_thread()`).
The packet is returned in partial chunks and printed.
The partial chunks may not be null terminated, but the length is known

Reply:
Ok. I see. `k_str_out` is marked as an internal API, so I'd just go with the `printk("%.*s"...)`  if it works with the default configuration of this sample.